[Bug]: NPE on JmsIO

### What happened?

I have a Null pointer exception when using JmsIO with qpid-jms-client:1.11.0.

Here messageId can be null and raise a Null pointer exception : https://github.com/apache/beam/blob/release-2.59.0/sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java#L698C21-L698C30

stacktrace

```java
Exception in thread "main" java.lang.RuntimeException: java.io.IOException: java.lang.NullPointerException
	at org.apache.beam.runners.direct.DirectRunner$DirectPipelineResult.waitUntilFinish(DirectRunner.java:385)
	at org.apache.beam.runners.direct.DirectRunner$DirectPipelineResult.waitUntilFinish(DirectRunner.java:345)
	at org.apache.beam.runners.direct.DirectRunner.run(DirectRunner.java:218)
	at org.apache.beam.runners.direct.DirectRunner.run(DirectRunner.java:67)
	at org.apache.beam.sdk.Pipeline.run(Pipeline.java:325)
	at org.apache.beam.sdk.Pipeline.run(Pipeline.java:310)
	at com.renault.datalake.dll.pipelines.Main.main(Main.java:174)
Caused by: java.io.IOException: java.lang.NullPointerException
	at org.apache.beam.sdk.io.jms.JmsIO$UnboundedJmsReader.advance(JmsIO.java:702)
	at org.apache.beam.runners.direct.UnboundedReadEvaluatorFactory$UnboundedReadEvaluator.startReader(UnboundedReadEvaluatorFactory.java:243)
	at org.apache.beam.runners.direct.UnboundedReadEvaluatorFactory$UnboundedReadEvaluator.processElement(UnboundedReadEvaluatorFactory.java:138)
	at org.apache.beam.runners.direct.DirectTransformExecutor.processElements(DirectTransformExecutor.java:165)
	at org.apache.beam.runners.direct.DirectTransformExecutor.run(DirectTransformExecutor.java:129)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.NullPointerException
	at org.apache.beam.sdk.io.jms.JmsIO$UnboundedJmsReader.advance(JmsIO.java:698)
	... 10 more
```

### Issue Priority

Priority: 2 (default / most bugs should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Infrastructure
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

--- a/sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java
+++ b/sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java
@@ -31,6 +31,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.NoSuchElementException;
 import java.util.Optional;
+import java.util.UUID;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 import java.util.stream.Stream;
@@ -586,7 +587,7 @@ public class JmsIO {
   }
 
   static class UnboundedJmsReader<T> extends UnboundedReader<T> {
-
+    private static final byte[] EMPTY = new byte[0];
     private UnboundedJmsSource<T> source;
     @VisibleForTesting JmsCheckpointMark.Preparer checkpointMarkPreparer;
     private Connection connection;
@@ -604,7 +605,7 @@ public class JmsIO {
       this.source = source;
       this.checkpointMarkPreparer = JmsCheckpointMark.newPreparer();
       this.currentMessage = null;
-      this.currentID = new byte[0];
+      this.currentID = EMPTY;
       this.options = options;
     }
 
@@ -684,18 +685,22 @@ public class JmsIO {
         currentTimestamp = new Instant(message.getJMSTimestamp());
 
         String messageID = message.getJMSMessageID();
-        if (this.source.spec.isRequiresDeduping()) {
-          // per JMS specification, message ID has prefix "id:". The runner use it to dedup message.
-          // Empty or non-exist message id (possible for optimization configuration set) will induce
-          // data loss.
-          if (messageID.length() <= 3) {
-            throw new RuntimeException(
-                String.format(
-                    "Invalid JMSMessageID %s while requiresDeduping is set. Data loss possible.",
-                    messageID));
+        if (messageID != null) {
+          if (this.source.spec.isRequiresDeduping()) {
+            // per JMS specification, message ID has prefix "id:". The runner use it to dedup
+            // message. Empty or non-exist message id (possible for optimization configuration set)
+            // will cause data loss.
+            if (messageID.length() <= 3) {
+              throw new RuntimeException(
+                  String.format(
+                      "Invalid JMSMessageID %s while requiresDeduping is set. Data loss possible.",
+                      messageID));
+            }
           }
+          currentID = messageID.getBytes(StandardCharsets.UTF_8);
+        } else {
+          currentID = EMPTY;
         }
-        currentID = messageID.getBytes(StandardCharsets.UTF_8);
 
         return true;
       } catch (Exception e) {
@@ -728,6 +733,12 @@ public class JmsIO {
     public byte[] getCurrentRecordId() {
       if (currentMessage == null) {
         throw new NoSuchElementException();
+      } else if (currentID == EMPTY && this.source.spec.isRequiresDeduping()) {
+        LOG.warn(
+            "Empty JMSRecordID received when requiresDeduping enabled, runner deduplication will"
+                + " not be effective");
+        // Return a random UUID to ensure it won't get dedup
+        currentID = UUID.randomUUID().toString().getBytes(StandardCharsets.UTF_8);
       }
       return currentID;
     }

